go/runtime/host/sgx/tcb_cache.go

```diff
@@ -0,0 +1,115 @@
+package sgx
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/oasisprotocol/oasis-core/go/common/logging"
+	"github.com/oasisprotocol/oasis-core/go/common/persistent"
+	"github.com/oasisprotocol/oasis-core/go/common/sgx/pcs"
+)
+
+const (
+	tcbCacheKey = "tcb_bundle_cache"
+
+	tcbCacheRefreshThreshold    = 14 * 24 * time.Hour
+	tcbCacheSlowRefreshInterval = 24 * time.Hour
+)
+
+type tcbBundleCache struct {
+	Bundle         *pcs.TCBBundle `json:"bundle"`
+	ExpectedExpiry time.Time      `json:"expected_expiry"`
+	LastUpdate     time.Time      `json:"last_update"`
+}
+
+type tcbCache struct {
+	serviceStore *persistent.ServiceStore
+	logger       *logging.Logger
+	now          func() time.Time
+}
+
+func (tc *tcbCache) check() (*pcs.TCBBundle, bool) {
+	var err error
+
+	// Check if we have a copy in the local store.
+	var stored tcbBundleCache
+	switch err = tc.serviceStore.GetCBOR([]byte(tcbCacheKey), &stored); err {
+	case nil, persistent.ErrNotFound:
+		// Not cached yet. Not an error.
+	default:
+		// Can't get it... an error, but we can still try downloading it.
+		if tc.logger != nil {
+			tc.logger.Warn("error checking common store for cached TCB bundle",
+				"err", err,
+			)
+		}
+	}
+
+	// Decide if the cache needs to be refreshed.
+	if err != nil {
```

Comment:
I think it would be clearer if this conditional is moved into the above switch? e.g.
```golang
switch err {
  case nil:
     // No error, continues below.
  case persistent.ErrNotFound:
    // Not cached yet, not an error, but needs refresh.
    return nil, true
  default:
    // Log the error and needs refresh.
    ...
    return nil, true
}
```